### ISSUE TITLE
Usage of `deprecated` as a decorator factory

### ISSUE THREAD

**Issue by Gobot1234** [microsoft/pyright#5953]
I'd like to implement a `@todo` decorator that reminds people without looking at the body of the function that functionality is not yet implemented. I've spoken to Jelle about this and he thinks that something like
```py
import typing_extensions

todo = typing_extensions.deprecated("This function is not implemented yet")

@todo
def foo():
    raise NotImplementedError()
```
should be detectable with a type checker. Would pyright be able to add support for this?

**Comment by erictraut** [comment]
This would require quite a bit of hackery to support. 

I'm going to close the feature request for now, but it's something I might consider adding support for in the future if there's sufficient interest (in the form of upvotes).

**Comment by Gobot1234** [comment]
How many upvotes are required for sufficient interest?

**Comment by erictraut** [comment]
This enhancement request has received enough upvotes to reopen for consideration.

I've implemented the requested support, and it will be included in the next release of pyright.

**Comment by erictraut** [comment]
This is included in pyright 1.1.369.
